packages/playground/src/components/manage_gateway_dialog.vue

```diff
@@ -240,7 +254,7 @@ export default {
         (props.vm.projectName.toLowerCase().includes(ProjectName.Fullvm.toLowerCase()) ? "fvm" : "vm") +
         grid.config.twinId;
       prefix.value = oldPrefix.value + props.vm.name;
-      subdomain.value = generateName({ prefix: prefix.value }, 4);
+      subdomain.value = generateName({ prefix: prefix.value }, 4).toLowerCase();
```

Comment:
the vm's name might have `_`, so will it be valid after the lowercase?

Reply:
The issue arises because the VM name can contain underscores, but domain names cannot. Previously, we didn’t check for underscores when generating a domain name based on the VM name.

I added a function that suggests a subdomain name by checking for underscores in the VM name and replacing them with an empty string.

Good catch!